<launch>

<arg name="input_dim"/>
<arg name="max_evals"/>

<arg name="out_path" default="out.pkl"/>
<arg name="prog_path" default="prog.pkl"/>
<arg name="resume" default="true"/>
<arg name="critic_service" default="evaluator/get_critique"/>
<arg name="verbose" default="true"/>
<arg name="random_seed" default="1"/>

<node pkg="optim" type="CMAOptimization.py" name="optimizer" output="screen" required="true">
    <param name="output_path" value="$(arg out_path)"/>
    <param name="load_path" value="$(arg load_path)" if="$(arg resume)"/>
    <param name="progress_path" value="$(arg prog_path)"/>
    <param name="critic_service" value="$(arg critic_service)"/>
    <param name="verbose" value="$(arg verbose)"/>
    <param name="random_seed" value="$(arg random_seed)"/>
    <param name="input_dim" value="$(arg input_dim)"/>
    <param name="convergence/max_evaluations" value="$(max_evals)"/>
    <rosparam>
        convergence:
            output_tolerance: 0.001
            output_history_tolerance: 0.001
            input_tolerance: 0.05
            max_evaluations: 1000

        initial_std_dev: 0.5
        input_lower_bound: -1
        input_upper_bound: 1
    </rosparam>
</node>

</launch>